nsq_api/service.py

```diff
@@ -0,0 +1,215 @@
+import sys
+import time
+import socket
+import json
+
+from tornado import gen, concurrent
+import tornado.ioloop
+import tornado.web
+import requests
+from nsq.reader import Reader, Message
+from basescript import BaseScript
+from deeputil import generate_random_string, AttrDict, keeprunning 
+from logagg_utils import NSQSender
+from logagg_utils import InvalidArgument, AuthenticationFailure
+from logagg_utils import start_daemon_thread, log_exception
+
+class NsqAPI(tornado.web.RequestHandler):
+
+    NSQ_MAX_IN_FLIGHT = 200
+    CHANNEL_NAME_LENGTH = 6
+    DELETE_CHANNEL_URL = 'http://{}/channel/delete?topic={}&channel={}'
+    executor = concurrent.futures.ThreadPoolExecutor(5)
+
+    def initialize(self, log):
+
+        self.log = log
+
+    def _parse_query(self, query_arguments):
+        '''
+        query_arguments: {'topic': ['Heartbeat'], 'nsqd_tcp_address': ['195.201.98.142:4150']}
```

Comment:
I don't think we should hardcode the IP address here.

Reply:
It's just a comment with arbitrary IP address.